### PR TITLE
Handle deprecated random module

### PR DESCRIPTION
**Pull request by nickva**
Use a compile time check for platform versions, then a macro conditional in a
separate rand module.

### CODE REVIEW

File: test/b64url_tests.erl [hunk 1]
```diff
@@ -85,31 +86,31 @@ bad_len_binary() ->
...
+        3->
             [First, Second]
     end.
 
 
 insert_error(B) when is_binary(B), size(B) < 2 ->
-    case crypto:rand_uniform(0, 2) of
-        0 -> {<<122, 255>>, 0};
-        1 -> {<<122, 122, 255>>, 0}
+    case b64url_rand:uniform(2) of
```
Comment (wohali) [inline test/b64url_tests.erl:107]: It's unexpected that our rand_uniform replacement doesn't have the same results as crypto:rand_uniform.

**Comment by wohali** [comment]
I'm -0 on the change in syntax/semantics for the fn call but otherwise looks good :)

File: test/b64url_tests.erl (hunk 1 above)
Comment (nickva) [reply test/b64url_tests.erl:107]: Rand crypto:rand_uniform was our previous attempt at replacing random and it does have different ranges of values. Our couch_rand module those does have same results

Comment (wohali) [reply test/b64url_tests.erl:107]: ok then +1 if we now match erlang!

**Pull request merged**